library/bignum.c

```diff
@@ -116,6 +120,7 @@ void mbedtls_mpi_free( mbedtls_mpi *X )
 int mbedtls_mpi_grow( mbedtls_mpi *X, size_t nblimbs )
 {
     mbedtls_mpi_uint *p;
+    MPI_VALIDATE_RET( X != NULL );
```

Comment:
Since the condition is that the MPI must be initialized, we could try asserting something a bit stronger, for example:
```
MPI_VALIDATE_RET( X != NULL && ( X->p != NULL || X->n == 0 ) );
```
This could even be turned to a macro called for example `MPI_VALIDATE_INIT_RET()` (taking just the name of the variable as argument).

I know we're usually content with checking non-NULLity, but here it's relatively easy to check more, especially as there's another pointer we're going to dereference in most of the functions, so why not check if it's NULL too?

(This holds for all occurrences obviously. Also in tests, we would test with bad MPIs in addition to testing with NULL arguments.)

Reply:
@mpg @sbutcher-arm I would like to stick with NULL checks only. Parameter validation is meant to check for _obvious_ violations of function preconditions, and while structural integrity definitely is such a precondition (except for `xxx_init()`), in my opinion it goes to far to start checking it in detail. If you start, where would you draw the boundary? Also, note that there are precondition we simply cannot effectively check, like buffer validity, so we must live with parameter validation being an approximation to the preconditions.

Reply:
@hanno-arm @sbutcher-arm In the interest of moving forward, I'd suggest we go with only (top-level) `NULL` checks for this PR, and if we opt to implement extra checks, postpone them to a later PR.